### PR TITLE
refactor Request#prepare!

### PR DESCRIPTION
**Pull request by technoweenie**
See https://github.com/lostisland/hurley/pull/5#issuecomment-68753005

### CODE REVIEW

**Comment by technoweenie** [comment]
FWIW: I hate the `prepare` name.  If anyone has better ideas on a convention for it, I'm all ears.  I recently added a `Hurley::Response#complete!` method too.  Both are intended to be called by internal hurley code or adapters, and not by users directly.

**Comment by pengwynn** [comment]
Of the `build`, `setup`, `prepare` usual suspects, I don't mind `prepare` here.

File: lib/hurley/client.rb [hunk 1]
```diff
@@ -176,40 +176,53 @@ def inspect
...
+        self.body = io
+      when Hash
+        ctype, io = options.build_form(body)
+        self.body = io
+      end
+      header[:content_type] ||= ctype || DEFAULT_TYPE
+    end
+
+    def prepare_content_length!
+      if header.key?(:content_length) || header[:transfer_encoding] == CHUNKED
```
Comment (Ben-M) [inline lib/hurley/client.rb]: why not make this look more like a guard clause?

return if header.key?(:content_length) || header[:transfer_encoding] == CHUNKED

Comment (Ben-M) [inline lib/hurley/client.rb]: prepare_basic_auth! and prepare_content_length! are ok, but why not be more descriptive? set_basic_auth_headers!, set_content_length_header! ?

I also wonder if prepare_body! is doing too much?

Comment (technoweenie) [inline lib/hurley/client.rb]: > why not make this look more like a guard clause?

That line is long enough as it is.

> why not be more descriptive? 

Not sure it's needed.  I do plan to go through and write Tomdoc comments for all the methods when we're done.  Just waiting for the dust to settle a bit with this v0.1 library.  

> I also wonder if prepare_body! is doing too much?

Body and content-type are pretty closely related.  Especially when you're converting some ruby object into a body.  I thought about making teaching `RequestOptions#build_form` how to handle Query objects.  All it's doing is converting a Hash to the Query type (Query::Nested or Query::Flat) the Request is using, and calling `#to_form` on that.  I also thought about renaming `#to_form` to something generic like `#prepare_body_and_type`.  Then Hurley can turn any ruby object into a body as long as it defines that method.  That could come from a future PR.

Comment (Ben-M) [inline lib/hurley/client.rb]: >Not sure it's needed. I do plan to go through and write Tomdoc comments for all the methods when we're done. Just waiting for the dust to settle a bit with this v0.1 library.

Descriptive method names are lower maintenance than Tomdoc comments. Either way, breaking it down is great, so let's merge it.

**Pull request merged**